### PR TITLE
Add Log Converter for ROS2 -> ReSim

### PR DESCRIPTION
**Pull request by mikebauer**
## Description of change
Add a Log Converter library that converts a ROS2 bag to a ReSim mcap log using a resim::msg::ConverterPlugin.

Prev PRs:
 - #30 
 - #29 

## Guide to reproduce test results.
```bash
full_test () {                                                                                      
   TARGET="$1"                                                                                      
   VALGRIND="valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes --error-exitcode=1"

   bazel test --runs_per_test=100 "${TARGET}" && \                                                  
   bazel test --test_timeout=3600 --run_under="${VALGRIND}" "${TARGET}"       
}
full_test //resim/msg:resim_log_from_ros2_test
```
## Checklist:
- [x] I have self-reviewed this change.
- [x] I have tested this change.
- [x] This change is covered by tests that are already landed, or in this PR.

### CODE REVIEW

File: resim/third_party/ros2/copts.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+--- a/ros2/cc_opts.bzl
++++ b/ros2/cc_opts.bzl
+@@ -1,5 +1,5 @@
+ """ Defines commonly used C/C++ options.
+ """
+ 
+-C_COPTS = ["-std=c11"]
+-CPP_COPTS = ["-std=c++17"]
++C_COPTS = []
++CPP_COPTS = []
```
Comment (mikebauer) [inline resim/third_party/ros2/copts.patch:12]: We need this patch to use C++20 when building ROS2 targets.

File: resim/msg/resim_log_from_ros2_test.cc [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+              resim::random_element<uint32_t>(InOut{rng});
+          REASSERT(result[channel.name]
+                       .emplace(
+                           stamp,
+                           std::make_unique<MessageType>(
+                               random_element<MessageType>(InOut{rng})))
+                       .second);
+        }
+        // Need to return something even though we don't use it.
+        return true;
```
Comment (knowlsie) [inline resim/msg/resim_log_from_ros2_test.cc:77]: Discussed IRL

File: resim/third_party/ros2/copts.patch (hunk 1 above)
Comment (knowlsie) [reply resim/third_party/ros2/copts.patch:12]: I'm about 90% confident this comment is obscuring a bunch more complexity, but I'll believe you.

**Review by knowlsie — APPROVED** [review]

File: resim/msg/resim_log_from_ros2_test.cc (hunk 2 above)
Comment (mikebauer) [reply resim/msg/resim_log_from_ros2_test.cc:77]: I'm going to leave this as tech debt for now, since it's a bit ugly to work around quickly.

**Pull request merged**